pandas/core/arrays/categorical.py

```diff
@@ -644,7 +644,11 @@ def from_codes(cls, codes, categories=None, ordered=None, dtype=None):
             )
             raise ValueError(msg)
 
-        codes = np.asarray(codes)  # #21767
+        if is_extension_array_dtype(codes) and is_integer_dtype(codes):
+            # Avoid the implicit conversion of Int to object
+            codes = codes.to_numpy(dtype=np.int64)
+        else:
+            codes = np.asarray(codes)  # #21767
```

Comment:
you don't need to add the issue number in code generally, unless you add a full expl (so here just remove i)